### PR TITLE
Feature/nm 68 mapping select team component to real data

### PR DESCRIPTION
**Pull request by Petko12345**

### CODE REVIEW

File: src/components/header/Header.tsx [hunk 1]
```diff
@@ -45,7 +51,7 @@ class Header extends React.Component<IProps, any> {
         <ViewSwitcher mode={ViewMode.tiles} />
 
         <div className={styles.header_selectTeamAndSort}>
-          <Selector {...mock.selectTeamProps} handleSelection={this.bind.onTeamSelect} />
+          <Selector label="Select Team" mode={SelectorMode.Basic} options={this.props.teams !== undefined ? this.createArrayOfTeamsForSelector(this.props.teams) : []} handleSelection={this.bind.onTeamSelect} />
```
Comment (dullus) [inline src/components/header/Header.tsx]: performance - avoid nonatomic conditions inside options prop. Do it outside JSX object and set constant.
```ts
const options = this.props.teams !== undefined ? this.createArrayOfTeamsForSelector(this.props.teams) : [];
return (
... options={options}
```

File: src/components/header/Header.container.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+import { fetchGit, fetchJira } from '../../actions/fetch';
+import { Header } from './Header';
+
+{/* @TODO temporary way how to show teams in dropdown. Shoudn't be used state.jira.teams*/ }
+const mapStateToProps = (state: any) => ({
```
Comment (dullus) [inline src/components/header/Header.container.tsx]: we have State interface in interfaces.ts

File: src/components/header/Header.tsx [hunk 3]
```diff
@@ -69,6 +75,16 @@ class Header extends React.Component<IProps, any> {
     );
   }
 
+  private createArrayOfTeamsForSelector(teams: any[]): ITeam[] {
+    var teamsForSelector: ITeam[] = [];
```
Comment (dullus) [inline src/components/header/Header.tsx]: var is obsolete

File: src/components/header/Header.tsx [hunk 4]
```diff
@@ -45,7 +51,7 @@ class Header extends React.Component<IProps, any> {
         <ViewSwitcher mode={ViewMode.tiles} />
 
         <div className={styles.header_selectTeamAndSort}>
-          <Selector {...mock.selectTeamProps} handleSelection={this.bind.onTeamSelect} />
+          <Selector label="Select Team" mode={SelectorMode.Basic} options={this.props.teams !== undefined ? this.createArrayOfTeamsForSelector(this.props.teams) : []} handleSelection={this.bind.onTeamSelect} />
           <Selector {...mock.sortByProps} handleSelection={this.bind.onSortTypeSelect} />
```
Comment (lubomir1618) [inline src/components/header/Header.tsx]: I think here is enough: this.props.teams  ? this.createArrayOfTeamsForSelector(this.props.teams) : []

File: src/components/header/Header.tsx [hunk 5]
```diff
@@ -69,6 +75,16 @@ class Header extends React.Component<IProps, any> {
     );
   }
 
+  private createArrayOfTeamsForSelector(teams: any[]): ITeam[] {
+    var teamsForSelector: ITeam[] = [];
+
+    teams.map((team: any, i: number) =>
+      teamsForSelector.push({ id: i, text: team.name })
```
Comment (dullus) [inline src/components/header/Header.tsx]: don't use push inside map loop, it returns array anyway. Whole fun body can be oneliner:
```
return teams.map((team, i) => ({ id: i, text: team.name }));
```

File: src/components/header/Header.tsx [hunk 6]
```diff
@@ -69,6 +75,16 @@ class Header extends React.Component<IProps, any> {
     );
   }
 
+  private createArrayOfTeamsForSelector(teams: any[]): ITeam[] {
```
Comment (dullus) [inline src/components/header/Header.tsx]: I think we can provide Team interface instead any. Use better naming, ITeam for feed data and ITeamSelect for <Selector> options.

File: src/components/header/Header.tsx [hunk 7]
```diff
@@ -3,17 +3,23 @@ import { connect } from 'react-redux';
...
-import Selector from '../selector/Selector';
+import Selector, { SelectorMode } from '../selector/Selector';
 import ViewSwitcher, { ViewMode } from '../viewSwitcher/ViewSwitcher';
 
 import styles from './Header.module.css';
 import * as mock from './Header.mock';
 
 interface IProps {
   isLogged?: boolean;
+  teams?: any;
```
Comment (dullus) [inline src/components/header/Header.tsx]: provide interface for teams

**Pull request merged**